### PR TITLE
M3-1828 search landing

### PR DESCRIPTION
**Pull request by Jskobos**
## Purpose

Users should be able to visit a dedicated search landing page, with bookmarkable URLs, that will display all of the results.

## Notes

1. Access the search results page by visiting `/search/?=query`, or by using the search bar and selecting the final option.
2. If navigating from the search bar, search results will be passed across to the landing page (via Router) to avoid network requests/filtering. If loading directly, requests are made from the landing page as they would be from the search bar.
3. Includes refactoring of search methods into Search/utils.ts, since the client-side filtering is the same in both the search bar and the search landing page.
4. Some logic and state is duplicated (see bullet 2 above), since functionality is not the same (e.g. the landing page only has to search once). It would be possible to add the full list of entities in Redux, for example, but since this is supposed to be a temporary solution until we have API search support, Redux is unnecessarily complicated.

### CODE REVIEW

File: src/features/TopMenu/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -395,20 +275,48 @@ class SearchBar extends React.Component<CombinedProps, State> {
...
+  onOpen = () => {
+    this.setState({ searchActive: true });
+  }
+
   onSelect = (item: Item) => {
     if (!item || isEmpty(item)) { return; }
     const { history } = this.props;
-    this.toggleSearch();
+    const { searchText } = item.data;
+    if (item.value === 'redirect') {
```
Comment (johnwcallahan) [inline src/features/TopMenu/SearchBar/SearchBar.tsx]: This is great! UX is so smooth.

File: src/features/Search/SearchLanding.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+interface TypesContextProps {
+  typesData?: Linode.LinodeType[];
+}
+
+type CombinedProps = TypesContextProps & RouteComponentProps<{}> & WithStyles<ClassNames>;
+
+export class SearchLanding extends React.Component<CombinedProps, State> {
+  mounted: boolean = false;
+  getQuery = () => {
```
[the remark this answers is not in the document]
Comment (rjstires) [reply src/features/Search/SearchLanding.tsx]: I recently found this as well, and I don't believe it's related to this PR. That said, there's 7,000 quality NPM packages out there that handle parsing querystrings.

File: src/features/Search/utils.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+)
+
+export const searchVolumes = (volumes: Linode.Volume[], query: string) => volumes.filter(
+  volume => volume.label.toLowerCase().includes(query),
+).map(volume => ({
+  label: volume.label,
+  value: volume.id,
+  data: {
+    tags: [],
+    description: volume.size + ' G',
```
[the remark this answers is not in the document]
Comment (rjstires) [reply src/features/Search/utils.ts]: And volumes are GiB.

File: src/features/Search/SearchLanding.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jskobos) [reply src/features/Search/SearchLanding.tsx]: This error is explicitly thrown by our parseQueryParams function. I wrapped it in a try/catch block for now, and am looking into libraries

**Pull request merged**